src/inmanta/deploy/state.py

```diff
@@ -102,6 +102,22 @@ class ModelState:
     ResourceStatus so that it lives outside of the scheduler lock's scope.
     """
 
+    def construct(
+        self,
+        resources: Mapping[ResourceIdStr, ResourceDetails],
+        requires: Mapping[ResourceIdStr, Set[ResourceIdStr]],
+    ) -> None:
+        """
+        Construct the current model state based on the provided parameters
+
+        :param resources: The resources that need to be added
+        :param requires: The `requires` field of these resources
+        """
+        for resource_id, details in resources.items():
+            self.update_desired_state(resource_id, details)
+        for resource_id, require_list in requires.items():
+            self.update_requires(resource_id, require_list)
```

Comment:
this has to go I think, the code already does this (but much more delicate)

Reply:
If the scheduler crashes, we need to have this information available when it's restarted even if a new version has not been released, right?